dbms/src/Flash/Mpp/ReceivedMessageQueue.h

```diff
@@ -55,6 +57,65 @@ enum class ReceiverMode
     Async
 };
 
+class GRPCNotifyQueue : public NotifyFuture
+{
+public:
+    template <typename... Args>
+    explicit GRPCNotifyQueue(const LoggerPtr & log_, Args &&... args)
+        : queue(log_, std::forward<Args>(args)...)
+    {}
+
+    void registerTask(TaskPtr && task) override { queue.registerPipeReadTask(std::move(task)); }
+
+    MPMCQueueResult pop(ReceivedMessagePtr & data) { return queue.pop(data); }
+
+    MPMCQueueResult tryPop(ReceivedMessagePtr & data) { return queue.tryPop(data); }
+
+    MPMCQueueResult forcePush(ReceivedMessagePtr && data) { return queue.forcePush(std::move(data)); }
+
+    MPMCQueueResult push(ReceivedMessagePtr && data) { return queue.push(std::move(data)); }
+
+    MPMCQueueResult tryDequeue() { return queue.tryDequeue(); }
+
+    MPMCQueueResult pushWithTag(ReceivedMessagePtr && data, GRPCKickTag * new_tag)
+    {
+        return queue.pushWithTag(std::move(data), new_tag);
+    }
+
+    void setKickFuncForTest(GRPCKickFunc && func) { queue.setKickFuncForTest(std::move(func)); }
+
+    bool finish() { return queue.finish(); }
+    bool cancel() { return queue.cancel(); }
+
+    bool isWritable() const { return queue.isWritable(); }
+
+    void registerPipeWriteTask(TaskPtr && task) { queue.registerPipeWriteTask(std::move(task)); }
+
+private:
+    GRPCRecvQueue<ReceivedMessagePtr> queue;
```

Comment:
How about directly letting `GRPCRecvNotifyQueue`  inherit `GRPCRecvQueue<ReceivedMessagePtr>`? If so, these functions do not need to be written, only the `registerTask` function is required.